File: javascript/IAdminService.ts
/// <reference path="./YaasModel" />
export interface IAdminService {
	change_password_for_user(request: org.roylance.yaas.UIYaasRequest, onSuccess:(response: org.roylance.yaas.UIYaasResponse)=>void, onError:(response:any)=>void)
	get_all_users(request: org.roylance.yaas.UIYaasRequest, onSuccess:(response: org.roylance.yaas.UIYaasResponse)=>void, onError:(response:any)=>void)
	is_user_admin(request: org.roylance.yaas.UIYaasRequest, onSuccess:(response: org.roylance.yaas.UIYaasResponse)=>void, onError:(response:any)=>void)
	set_user_as_admin(request: org.roylance.yaas.UIYaasRequest, onSuccess:(response: org.roylance.yaas.UIYaasResponse)=>void, onError:(response:any)=>void)
	remove_user_as_admin(request: org.roylance.yaas.UIYaasRequest, onSuccess:(response: org.roylance.yaas.UIYaasResponse)=>void, onError:(response:any)=>void)
	delete_user(request: org.roylance.yaas.UIYaasRequest, onSuccess:(response: org.roylance.yaas.UIYaasResponse)=>void, onError:(response:any)=>void)
}

File: csharp/Org.Roylance.Yaas.Api/IAdminService.cs
using System.Threading.Tasks;

namespace Org.Roylance.Yaas.Api
{
    public interface IAdminService
    {
		Task<Org.Roylance.Yaas.UIYaasResponse> change_password_for_user(Org.Roylance.Yaas.UIYaasRequest request);
		Task<Org.Roylance.Yaas.UIYaasResponse> get_all_users(Org.Roylance.Yaas.UIYaasRequest request);
		Task<Org.Roylance.Yaas.UIYaasResponse> is_user_admin(Org.Roylance.Yaas.UIYaasRequest request);
		Task<Org.Roylance.Yaas.UIYaasResponse> set_user_as_admin(Org.Roylance.Yaas.UIYaasRequest request);
		Task<Org.Roylance.Yaas.UIYaasResponse> remove_user_as_admin(Org.Roylance.Yaas.UIYaasRequest request);
		Task<Org.Roylance.Yaas.UIYaasResponse> delete_user(Org.Roylance.Yaas.UIYaasRequest request);
	}
}
